File: Euclid/Messages/Outgoing/SECRET_KEY.cs
using Euclid.Network.Streams.Util;

namespace Euclid.Messages.Outgoing
{
    class SECRET_KEY : IMessageComposer
    {
        private string publicKey;

        public SECRET_KEY(string publicKey)
        {
            this.publicKey = publicKey;
        }

        public override void Write()
        {
            this.Data.Add(new ArgumentEntry(this.publicKey));
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/SECRET_KEY.java
package euclid.messages.outgoing;

import euclid.messages.IMessageComposer;
import euclid.network.streams.util.ArgumentEntry;

public class SECRET_KEY extends IMessageComposer {

    private final String publicKey;

    public SECRET_KEY(String publicKey) {
        this.publicKey = publicKey;
    }

    @Override
    public void write() {
        getData().add(new ArgumentEntry(publicKey));
    }
}
